src/sysfs_fuse.c

```diff
@@ -126,6 +127,93 @@ static int sys_devices_system_cpu_online_read(char *buf, size_t size,
 	return total_len;
 }
 
+static int filler_sys_devices_system_cpu(const char *path, void *buf,
+		fuse_fill_dir_t filler)
+{
+	__do_free char *cg = NULL, *cpuset = NULL;
+	struct fuse_context *fc = fuse_get_context();
+	pid_t initpid;
+
+	initpid = lookup_initpid_in_store(fc->pid);
+
+	if (initpid <= 1 || is_shared_pidns(initpid))
+		initpid = fc->pid;
+	cg = get_pid_cgroup(initpid, "cpuset");
+	if (!cg)
+		return 0; 
+
+	prune_init_slice(cg);
+
+	cpuset = get_cpuset(cg);
+	if (!cpuset)
+		return 0; 
+
+	char *tok;
+	lxc_iterate_parts(tok, cpuset, ",") {
+		if (strstr(tok, "-")) {
+			char *cpu[2];
+			int i = 0;
+
+			char *t;
+			lxc_iterate_parts(t, tok, "-") {
+				if(i > 1)
+					return 0;
+				cpu[i++] = t;
+			}
+
+			for(i=atoi(cpu[0]); i<=atoi(cpu[1]); i++) {
+				char cpu_str[10];
+				sprintf(cpu_str, "cpu%d", i);
+				if (DIR_FILLER(filler, buf, cpu_str, NULL, 0) != 0) {
+					return -ENOENT;
+				}
+			}
+		}
+		else {
+			char cpu_str[10];
+			sprintf(cpu_str, "cpu%d", atoi(tok));
+
+			if (DIR_FILLER(filler, buf, cpu_str, NULL, 0) != 0)
+				return -ENOENT;
+		}
+	}
+
+	__do_closedir DIR *dir = NULL;
+	struct dirent *dirent;
+	dir = opendir(path);
+	if (!dir)
+		return -ENOENT;
+
+	while ((dirent = readdir(dir))) {
+		if (strstr(dirent->d_name, "cpu") && 
+                dirent->d_name[strlen(dirent->d_name)-1] >= '0' &&
+                dirent->d_name[strlen(dirent->d_name)-1] <= '9')
+			continue;
+
+		if (DIR_FILLER(filler, buf, dirent->d_name, NULL, 0) != 0)
+			return -ENOENT;
+	}
+
+	return 0;
+}
+
+static mode_t get_st_mode(const char *path)
+{
+	struct stat sb;
+	__do_close int fd = -EBADF;
```

Comment:
Please place the cleanup macros as the first variable declaration in the function so it's easy to see what variables are scope-cleaned and which need to be manually cleaned (if any).